src/nvim/eval/funcs.c

```diff
@@ -7044,7 +7044,7 @@ static void f_resolve(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     xfree(buf);
   }
 # else
-  rettv->vval.v_string = (char_u *)xstrdup(p);
```

Comment:
Might be better to try using `os_realpath()` first (sort of like in the WIN32 case) and then fallback to just using the passed in value.

Reply:
Gonna have to try what's available, it's very much a non-standard system.

Reply:
I mean something like:

```c
char *v = os_realpath(fname);
rettv->vval.v_string = (char_u *)(v == NULL ? xstrdup(fname) : v);
```

That way we at least give libuv a chance to help us out.

Reply:
I tried out the `HAVE_READLINK` part and `os_realpath` separately, both seem to work. Termux also reports `-- Looking for readlink - found`.

Reply:
>  Termux also reports `-- Looking for readlink - found`.

Hmm, if that's the case and you're still getting into this part of the code, then something is wrong.  What does `grep READLINK build/config/auto/config.h` show?

Reply:
Everything looks good:
```
$ grep READLINK build/config/auto/config.h
#define HAVE_READLINK
```
I'm also not able to reproduce the original error anymore - the latest `master` branch builds fine. Have no idea what's going on anymore.

Reply:
Ok, well either way this is an obvious fix for places where `readlink` isn't available.